### PR TITLE
HDDS-4166. Documentation index page redirects to the wrong address

### PR DESCRIPTION
**Pull request by cku328**
## What changes were proposed in this pull request?

The server for the edge version of the documentation doesn't appear to use `index.html` as the default index page, but instead displays a list of directories. 

If the file name of the page is `index.html`, it doesn't add `index.html` to the `Permalink` parameters, which seems to be hugo's style. I found similar cases on the internet. (ref. [https://discourse.gohugo.io/t/ugly-urls-with-page-resources/10037](https://discourse.gohugo.io/t/ugly-urls-with-page-resources/10037))

The easiest solution is to check the permalink suffix and if it's not `.html` then print `index.html` at the end.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4166

## How was this patch tested?

Tested locally using `hugo serve`

擷取

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @cku328 for fixing this.

All _Home_ links in breadcrumbs (eg. in _Home > Features > Topology awareness_) are also affected, ie. `index.html` might need to be added to:

https://github.com/apache/hadoop-ozone/blob/13fe31b4927660bc2534656c6a7048b45d3c8051/hadoop-hdds/docs/themes/ozonedoc/layouts/_default/single.html#L33

FYI this can be reproduced locally by opening compiled docs from disk after `mvn package` without `hugo serve`, since there is no webserver in this case that would serve `index.html` as directory listing.

**Comment by cku328** [comment]
Thanks @adoroszlai for your review and comments.
I re-checked the hyperlinks in the document related to "index.html".

I will update this patch to include the following changes:
1. Fixed ozone logo image not displaying properly and added a hyperlink to the index page.
2. Added "index.html" to the "Home" hyperlink in the breadcrumb.
3. Added "index.html" to the "Overview" hyperlink in the sidebar.

**Review by adoroszlai — APPROVED** [review]
Thanks @cku328 for updating the patch.  I couldn't find any remaining non-http links ending in `/`.

**Comment by cku328** [comment]
Hi! @adoroszlai 
I found some broken hyperlinks and updated the patch to fix it.

It fixed the following items:
1. `Storage Container Manager` hyperlink in `/zh/concept/overview.html` page is not available.
2. `kerberos keytab` hyperlink in `/zh/security/secureozone.html`page is not available.
3. Replacing `ofs.md` with a hyperlink of OFS in `/zh/interface/o3fs.html`.

File: hadoop-hdds/docs/content/interface/O3fs.zh.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ hdfs dfs -ls o3fs://bucket.volume.om-host.example.com:6789/key
 注意：在这种情况下，`ozone.om.address` 配置中只有端口号会被用到，主机名是被忽略的。
 
 ## ofs的配置及使用
-这只是一个通用的介绍。了解更详细的用法，可以请参考ofs.md。
+这只是一个通用的介绍。了解更详细的用法，可以请参考[OFS(英文页面)]({{< relref path="interface/ofs.md" lang="en" >}})。
```
Comment (cxorm) [inline hadoop-hdds/docs/content/interface/O3fs.zh.md:118]: nit
```suggestion
这只是一个通用的介绍。了解更详细的用法，请参考[OFS(英文页面)]({{< relref path="interface/ofs.md" lang="en" >}})。
```

**Comment by cku328** [comment]
Thanks @cxorm for the review.
I've updated the patch based on your review.

File: hadoop-hdds/docs/content/interface/O3fs.zh.md (hunk 1 above)
Comment (cku328) [reply hadoop-hdds/docs/content/interface/O3fs.zh.md]: done

**Comment by adoroszlai** [comment]
Thanks @cku328 for the improvement and @cxorm for the review.  I have confirmed that @cxorm's wording suggestion is applied, so merging now.

**Pull request merged**